src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs

```diff
@@ -313,12 +438,36 @@ private bool VerifyAssetsFileMatchesProject()
             return (_request.ExistingLockFile != null && pathComparer.Equals( _request.ExistingLockFile.PackageSpec.FilePath , _request.Project.FilePath));
         }
 
-        private KeyValuePair<CacheFile, bool> EvaluateCacheFile()
+        private bool ValidatePackages(NuGetLockFile lockFile, LockFile assetsFile)
```

Comment:
This should have a better name.

Reply:
Yeah that might work. imho the current name is too simple. 

I'm not against making it even more verbose such as `ValidateLockFilePackagesShaMatchesAssetsFile`